transport.go

```diff
@@ -13,9 +13,29 @@ type WrapErrorFunc func(error) interface{}
 type Transporter interface {
 	getDispatcher() (dispatcher, error)
 	getReceiver() (receiver, error)
+
+	// Done returns a channel that's closed when incoming packets
+	// have finished processing, either due to an error or the
+	// underlying connection being closed. Successive calls to
+	// Done return the same value. This is safe to call before
+	// incoming packets have started processing (i.e., before
+	// Run() or RunAsync() has been called).
+	Done() <-chan struct{}
+
+	// Err returns a non-nil error value after Done is closed.
+	// After Done is closed, successive calls to Err return the
+	// same value.
+	Err() error
+
+	// IsConnected returns false when incoming packets have
+	// finished processing. This is exactly equivalent to checking
+	// if Done() is closed.
+	//
+	// TODO: Use a better name.
+	IsConnected() bool
```

Comment:
I'm not wild about IsConnected() anymore, since 'connected' seems different from 'stopped processing incoming packets'. But I'll fix this later to minimize churn.